presto-hive/src/main/java/com/facebook/presto/hive/DwrfEncryptionMetadata.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import static java.util.Objects.hash;
+import static java.util.Objects.requireNonNull;
+
+public class DwrfEncryptionMetadata
+        implements EncryptionMetadata
+{
+    private final Map<String, byte[]> fieldToKeyData;
```

Comment:
This should be changed to a fieldReference https://github.com/prestodb/presto/pull/14621/files#diff-bbdd3443793c8f1846fcec59930191df.  But if you want it can be changed later so there's no dependency between the PRs.

Reply:
`HiveColumnHandle` seems very heavy for this use case, Given that this information will be attached with every split I wanted it to be light. As of now, can we just not convert it back to a fieldReference using the information in `HiveTableLayoutHandle` and using `ColumnWithStructSubfield`

Reply:
Sure, that makes sense